### ISSUE TITLE
Consider moving chat off Freenode

### ISSUE THREAD

**Issue by jgraham** [whatwg/meta#210]
Freenode's ownership has [changed](https://gist.github.com/joepie91/df80d8d36cd9d1bde46ba018af497409) and it's unclear if it will remain a good venue to host the WHATWG chat.

If we move there are several possible options (implictly assuming the constraint that it should remain free):
* IRC through libera.chat. This seems like the most similar setup to what we already have.
* Matrix either through matrix.org or Mozilla's server
* Probably loads of others; there isn't a shortage of messaging platforms keen to offer free service at small scale.

One advantage of moving off IRC might be better tools for moderation and better history without requiring a tool like IRCCloud.

**Comment by domenic** [comment]
Although the ownership change sounds unsavory, I'm unsure if the impacts of the new ownership on WHATWG community members would be negative. Does anyone have information on what the actual downside of the new ownership is? I personally don't recall sharing any information with Freenode besides an auto-generated password for NickServe.

**Comment by jgraham** [comment]
I think the direct concern is that freenode.net either stops operating an IRC service or becomes less reliable. If I understand correctly all the people running the infrastructure have quit, so I think the medium term future of the service is up in the air, particuarly if there's a mass-exodus.

**Comment by sideshowbarker** [comment]
I agree the *“all the people running the infrastructure have quit*” fact seems to suggest we may have a relatively acute/urgent need to consider getting some other place set up where we can reconvene chat for the community.

And while it’s great to see how quickly libera.chat came together, maybe we should take the opportunity to really consider moving on to something other than IRC.

And as far as alternatives to IRC, it seems like in practice the main practical alternative other than Matrix is Slack.

I don’t feel strongly either way about Matrix vs Slack. I know Matrix is a great place for discussions with Mozilla devs and with people working on MDN. As far as Slack, I guess we should note here that chat for both the Chromium and WebKit projects now takes place on Slack instances they set up (after migrating away from Freenode IRC last year), and along with those, W3C now has a Slack instance too — as does the Open Web Docs project (where there’s now probably at least as much MDN discussion going on as there is in the MDN room on Matrix, and probably will be a lot more so going forward).

So I suppose that all seems to suggest that setting up a WHATWG Slack instance would be the thing that aligns most closely with where most of our closest peer communities are these days.

---

A few more thoughts/details:

As far as the reliability risk with Freenode, it seems like without the infrastructure continuing to be maintained by people who understand it, the timeframe for it becoming much less reliable is possibly just days or a few weeks. It sounds like kind of an “operating on a wing and a prayer” state (or whatever the appropriate metaphor is).

There’s also the fact we’ve occasionally had problems with spammers/abusers on the channel (and Freenode as a whole has had even worse problems with that) — and as far as I know the only reason those problems were mitigated was through the efforts of the people running the infrastructure doing some abuse control measure. And so without them around any longer, it seems likely it may just be a short time before we start to have more serious spam/abuse problems on the channel again.

Further, given the circumstances — that essentially a hostile takeover seems to have happened — I would imagine there’s also the risk of Freenode now getting hit with some genuine attacks on scale (botnet/DDOS stuff), intentionally designed to bring it down or at least make it as unusable as possible.

And while it sounds like libera.chat has the right people on board already to keep it relatively reliable and usable, I think we have a responsibility to consider the known concerns with IRC itself as a place for discussion — concerns which Mozilla documented in their decision to move away from IRC and over to Matrix.

IRC just does not seem to provide us with the same capabilities that Matrix and Slack and such do for mitigating abuse and for helping shield users of our community from abusive behavior. And so given that, deciding to switch to libera.chat or any other IRC-based mechanism at this point would not seem to be the responsible choice as far as giving the safety of community members the kind of consideration we should be.

**Comment by hsivonen** [comment]
On the Matrix vs. Slack issue, Matrix seems better from the dogfood point of view of creating standards for multiple interoperable Web engines. (Slack really likes to push users away from browser and has at least one single-engine feature that could be multi-engine.)

**Comment by sideshowbarker** [comment]
> On the Matrix vs. Slack issue, Matrix seems better from the dogfood point of view of creating standards for multiple interoperable Web engines. (Slack really likes to push users away from browser and has at least one single-engine feature that could be multi-engine.)

Yeah, I’m starting to understand and appreciate the thought that led Mozilla to end up choosing Matrix over the alternatives.

In particular, as far as Slack: after posting my previous comment, I spent a little time looking further into Slack — including trying to figure out how we could set up a Slack “workspace” (which I’ve never done), and how we could make any Slack workspace discoverable once we had set it up. And the outcome of that is, I found it pretty frustrating to find the info I was looking for. And I also found that stuff in Slack space is inherently not easily discoverable. The Slack model seems to silo/fragment “workspaces” into their own little islands all over the place, with no connections among them.

The Slack model seems to be built for the enterprise market — optimized for enterprise needs to the detriment of broader ones. (I suppose that’s probably already well-known, and the name “workspace” should have clued me in to that…)

To put it in other terms, Slack really doesn’t seem like an ideal tool for building or fostering open communities — it seems best for the case where you want to have isolated communities where newcomers aren’t actually expected to wander in very easily.

Matrix in contrast seems like a better home for a community where newcomers are expected to show up, and encouraged to.

And since the \#whatwg channel has been a gateway through which a lot of active contributors to WHATWG work over the years have first come in, preserving as much as we can of the characteristics of openness and discoverability it has would be useful in helping ensure we still have a good place for future newcomers to wander in and end up getting actively involved. And Matrix seems a lot better suited for making that happen than Slack does. (A libera.chat IRC channel  would clearly also have good openness and discoverability, but with the downsides of openness to abuse, and inherent difficulties in dealing with abuse when it happens — among other things that I guess were taken into consideration when Mozilla moved to Matrix.)

So after spending a modest amount of time looking into details of the alternatives today, I’m now hoping we put the strongest consideration into Matrix — or at least that we don’t just follow the trend into Slack that some of our peer communities have followed, and end up contributing to further silo-ization/fragmentation into Slack.

**Comment by sideshowbarker** [comment]
I guess we should also be taking Discord into consideration. I have zero experience personally with Discord, but I’ve so far gotten the impression that it might have some possible downsides (some of which are like the IRC downsides — or at least like some particular IRC community characteristics) that might not make it such a great choice for us.

**Comment by jgraham** [comment]
It appears that W3C IRC is now [bridged on matrix.org](https://github.com/matrix-org/matrix-appservice-irc/wiki/Bridged-IRC-networks) which seems like it could be a point in favour of matrix if you can use it as a complete replacement for IRC when doing web standards work.

**Comment by GPHemsley** [comment]
Consider me:
* Slack 👎🏻 
* Zulip 👎🏻 
* Discord 👎🏻 
* Gitter 👎🏻 

I know nothing about Matrix (and did not follow Mozilla once it left IRC), but I see the main benefits of IRC as being the fact that there is an open protocol that is widely supported, allowing folks to use whatever app they want to connect and not have to be beholden to some proprietary space, as well as the simplicity in its chat functionality.

Slack does not meet either of those requirements, and, IIUC, Discord doesn't either. (Plus, Discord is really still geared primarily to the gaming community.) Zulip is basically just open source Slack (IMHO), so doesn't really meet the requirements either. I don't know too much about Gitter, other than that any time I've encountered a project that uses it, it's basically a ghost town.

**Comment by jgraham** [comment]
I also get on well with the "bring your own client" nature of IRC. But I think those of us who do are in a minority, and the overall balance of arguments is strongly in favour of moving away from IRC:

* The properties of IRC aren't well understood by people who aren't used to it. For example I've tried to mentor outreachy applicants who were surprised that they were unable to disconnect without losing history. Adding that feature to IRC either requires extensive setup or using a third party service like IRCCloud (which are typically paid).
* IRC has relatively poor tools for moderation. We've occasionally had problems with spam, and I understand that people who are likely to be the target of abuse or hassasment online often avoid IRC. If we are knowingly using a service that is unwelcoming to some groups, that would arguably fall foul of the commitment in the participant agreement to provide a safe working environment.

Given those two points I think remaining on IRC is probably the wrong choice, and may be harmful to the long term future of the project.

**Comment by domenic** [comment]
I'm pretty against Matrix based on my experiences with it. If we're going for open protocol, let's stick with IRC.

If we want to move to a closed protocol, Discord is the most user-friendly choice in my experience between Discord and Slack. But both are OK.

I'm still pretty strongly on the side that we should stick with the current Freenode solution until/unless something bad happens. I personally don't plan to move before then.

**Comment by GPHemsley** [comment]
> @jgraham 
> * The properties of IRC aren't well understood by people who aren't used to it. For example I've tried to mentor outreachy applicants who were surprised that they were unable to disconnect without losing history. Adding that feature to IRC either requires extensive setup or using a third party service like IRCCloud (which typically require payment).
> * IRC has relatively poor tools for moderation. We've occasionally had problems with spam, and I understand that people who are likely to be the target of abuse or hassasment online often avoid IRC. If we are knowingly using a service that is unwelcoming to some groups, that would arguably fall foul of the commitment in the participant agreement to provide a safe working environment.
> 
> Given those two points I think remaining on IRC is probably the wrong choice, and may be harmful to the long term future of the project.

I think these two are salient points that we should not ignore.

I personally maintain my own logs of channels while I'm in them, so I at least have the history of what I've seen and have it restored between sessions, but I have also used the various log archives that are available for the WHATWG channel specifically to facilitate searches. I also am not clear on what moderation tools IRC lacks that are available elsewhere, but that's primarily because I haven't taken the time to educate myself on the details. (I'm sure Mozilla documented this thoroughly during their migration to Matrix. And I do know that Slack in particular lacks _any_ moderation tools, even those available in IRC.)

But none of that matters, because I'm not the audience we would be trying to solve for: I'm already here, and I'm already using IRC.

> @domenic 
> I'm still pretty strongly on the side that we should stick with the current Freenode solution until/unless something bad happens.

If this is the option we take (and I'm not clear that it's where the consensus is heading, but I've only just found out about these issues today), then I think we should instead take the opportunity to properly evaluate all our options in an attempt to maintain the good of what we have and improve along the lines described above.

**Comment by Krinkle** [comment]
Note that IRC and Matrix could live side-by-side fairly well (yes, I'm aware of imperfections and gotchas with bridging). This would allow people to use a free and modern hosted service with web app, or desktop/mobile app (via <https://app.element.io/>, or [Element apps](https://element.io/features)); whilst also allowing all existing users to connect via their current IRC clients and server (e.g. Freenode, or Libera Chat, or both!).

I'm personally not (yet) super fond of Matrix. It feels a bit quirky to my habituated IRC fingers. But, for a lot of people I think it significantly easier to get into. There are significantly more IRC clients out there than Matrix/Element clients, and so I'm sure somewhere out there is a great client for everyone, including many that have broken away from the traditional IRC look-n-feel. But, finding that can be quite a challenge for a newcomer. The small landscape around Matrix, on average, including Element's own apps, seem to be good a good starting off point.

Based on how things evolve in the coming years, we could then transition to another platform from Matrix, or keep supporting both, or transition to only Matrix if/when it feels "good" for all stakeholders.

There's a number of other projects I've seen switch to, or support, Matrix in this way (MediaWiki and Pixelfed from Freenode, and jquery-dev from Gitter; e.g. Freenode `#pixelfed` / <https://matrix.to/#/+pixelfed:matrix.org>).

**Comment by annevk** [comment]
I'm personally somewhat fed up with the spam on IRC and the lack of easy-to-use moderation tools. We're using Matrix and Slack at work and both are fine. WHATWG SG is currently using Slack and W3C seems to use it as well (at least in the privacy space). Big downside of Slack in my opinion is that you essentially end up with a tab for each Slack workspace. I like the Slack emoji reactions best though.

**Comment by jgraham** [comment]
I think the point about slack being mainly designed around a closed-community workspace is an important one. Anything that requires specific invites or is limited-seat is a significant barrier to open participation. That seems like more of a problem for WHATWG than W3C (the SG being an exception of course because it's explicitly closed membership).

**Comment by hsivonen** [comment]
Regarding Discord: Previously, they had terms of use that prohibited work use. I think that's now probably fixed, but now there appear to be [some other limits](https://twitter.com/gcpascutto/status/1395301406060462092). I haven't investigated this properly, but these things that have anecdotally come to my attention about Discord terms make me think: Why deal with that stuff when there are other options?

Regarding Zulip: It's used for Rust compiler work. It's a rather strange mix of chat and message board. Maybe that's good if you get used to it, but it sure is _weird_ when looking for an IRC replacement.

(I agree with what @jgraham said in the previous comment about closed workspace vs. open for participation.)

**Comment by andreubotella** [comment]
It seems like [LogBot is shutting down soon](https://freenode.logbot.info/_shutdown). Although there's no mention of why it's shutting down, it's almost certainly a result of this hostile takeover. I believe this gives us one more reason to switch services.

In any case, regardless of whether we move the chat to somewhere else or find another logging bot, whenever LogBot publishes the `#whatwg` archives we should probably host them, as with [lists.whatwg.org](https://lists.whatwg.org/).

**Comment by foolip** [comment]
Are there #whatwg IRC log archives somewhere now that we can make a copy of?

**Comment by sideshowbarker** [comment]
> Are there #whatwg IRC log archives somewhere now that we can make a copy of?

https://freenode.logbot.info/_shutdown says:

> Once the IRC bots have stopped running I will make archives of each channel's logs available for download.

But I reckon we might at any time be able to get an archive of the logs so far, if somebody just asks glob if he can make us an archive.

**Comment by Krinkle** [comment]
I don't know about network stability, but please note that Freenode's new manager has escalated their harrasment compaign to the official homepage at <https://freenode.net/>, where their latest blogpost spreads various falsehoods and misattributes words by contributors from Rsync, Wikipedia, and other projects. 

Here's a guide with some tips:
<https://meta.wikimedia.org/wiki/IRC/Migrating_to_Libera_Chat>

**Comment by jgraham** [comment]
It has also been widely reported that the network admins have taken over several hundred channels for the "policy violation" of mentioning other IRC networks in the /topic.

I don't feel able to tell would-be contributors to join Freenode at this point. I fear that if we just let this slide instead of making a definite decision, we're not only implictly endorsing the way the network is being run, but also valuing a minor inconvenience for people already working on WHATWG standards above our ability to attract new people and keep the project healthy.

@whatwg/sg what's the process to reach a conclusion here? I think the scope of the question is "which synchronous communication channel(s) should we advertise as the official on whatwg.org and consider moderated according to the code of conduct?".

**Comment by annevk** [comment]
So based on the discussion so far it seems the more serious options are:

1. Matrix
2. IRC (Libera)

Is that correct?

Can we do both by setting up the matrix channel as a bridge?

(I'm not sure the SG is best suited to make a decision here, but perhaps it's a good escalation point if we reach some kind of impasse.)

**Comment by jgraham** [comment]
I believe it's not currently possible to bridge Libera from matrix, but there's a [process underway](https://github.com/matrix-org/matrix-appservice-irc/issues/1324) which will make it possible soon.

I think my argument against bridging is that it adds an additional layer of complexity which might be unnecessary. I expect we'd point new users to matrix before suggesting they use IRC since that's much more like the majority of other chat services they're likely to be using. It's also unclear to me (and maybe someone who already bridges to the FreeNode channel can comment) how it affects the functionality of matrix. Do we lose features like reaction emoji (and maybe in the future threads) which don't map 1:1 with IRC? Also, once logbot stops, you'll need to use matrix for logs anyway (unless someone else stands up a logger, or you're using a paid service with built-in logging). So whilst I see that "IRC with an officially endorsed bridge" feels like an easy transition path, I'm not sure it's actually the best long term decision.

**Comment by jgraham** [comment]
` <Bakkot> annevk tc39 is officially moving to matrix, we've just decided`

**Comment by andreubotella** [comment]
I've tried out a few of the Matrix features that don't map to IRC on the `#freenode_#whatwg:matrix.org` bridge (which is powered by [matrix-appservice-irc](https://github.com/matrix-org/matrix-appservice-irc)), and Matrix will let you use those features just fine – it's just that from the IRC side you might not have the best experience.

- Custom emoji can be added and removed without any traffic to the IRC side.
![image](https://user-images.githubusercontent.com/8225977/119723389-ad1cf180-be6d-11eb-9e03-53ddfb7eb245.png)

- Editing a message will result in the edited version being sent on the IRC side, with an asterisk in front.
![image](https://user-images.githubusercontent.com/8225977/119723566-e2294400-be6d-11eb-9bbb-147b37ca8a4b.png)
![image](https://user-images.githubusercontent.com/8225977/119723910-5532ba80-be6e-11eb-8589-1aa7319772cf.png)

- Deleting a message from Matrix will do nothing on the IRC side.
![image](https://user-images.githubusercontent.com/8225977/119724030-772c3d00-be6e-11eb-9117-e5fe913b5850.png)
![image](https://user-images.githubusercontent.com/8225977/119724064-827f6880-be6e-11eb-937e-057b842ab153.png)

- Quoting a message will prepend a shortened version of the quoted message.
![image](https://user-images.githubusercontent.com/8225977/119724196-b2c70700-be6e-11eb-985a-3f39dc46c29e.png)
![image](https://user-images.githubusercontent.com/8225977/119724230-bfe3f600-be6e-11eb-9567-ea72c8486f48.png)

**Comment by MayeulC** [comment]
@jgraham 
> `<jgraham >So the freenode_ one is presumably a bridge to this channel. I think nox's objection to the :mozilla.org one is reasonable (or at least I think the concern that the channel should be in a vendor-neutral space is reasonable)`

Since most people here do not seem to be familiar with how matrix works, allow me to point out something: On Matrix, a room exists simultaneously on every server. It's only identified by its room id, which is expected to be globally unique. New messages propagate to all participating servers (and are checked against existing rules: a mod can't demote an admin, for instance).
Room IDs are unwieldy, and you need to know at least one participating server to join it from a new server, so there is also the concept of room aliases. Anyone can create aliases on their server to point to the room, although one alias can be chosen as the "canonical" one.

Long story short: `whatwg:matrix.org` and `whatwg:mozilla.org` (and :ponies.im and :kde.org and whatnot) can all lead to the same room. Server administrators can also override aliases if someone squats them.

About clients, some third-party clients work quite well already. I can mention [fluffychat](https://fluffychat.im/), [nheko](https://nheko.im/nheko-reborn/nheko) and [gomuks](https://github.com/tulir/gomuks): I tend to use them more often than Element. Also, [neochat](https://apps.kde.org/neochat/) has progressed quite well lately.

**Comment by annevk** [comment]
Update: a Matrix room has been established, https://whatwg.org/chat has been updated with details. For now it still mentions IRC details.

**Comment by GPHemsley** [comment]
There are now 24 of us (and counting) in the #whatwg channel on Libera Chat. However, the channel is not currently registered and has no topic.

Given how things are escalating on the Freenode side, I think it would be a good idea to get stuff moving on the Libera Chat side, even if Matrix becomes the preferred option for newcomers.

Have any efforts yet been made to make that happen? (I don't know if we'd want to register an entire project, but there is a backlog if we haven't gotten in on that yet.)

**Comment by annevk** [comment]
I think if Freenode falls we should use Matrix exclusively, especially now that TC39 has picked it as well.

**Comment by Krinkle** [comment]
Matrix now has a first-party bridge to Libera, suggesting that this is doable now.

<https://twitter.com/matrixdotorg/status/1398366903404482564>

**Comment by andreubotella** [comment]
Just as a reminder, these are the current chats and their Matrix bridges:

- [#whatwg on Freenode](ircs://irc.freenode.net/whatwg) is still a thing for the time being. It's bridged on Matrix at [#freenode_#whatwg:matrix.org](https://matrix.to/#/#freenode_#whatwg:matrix.org)
- The WHATWG room at matrix.org is [#whatwg:matrix.org](https://matrix.to/#/#whatwg:matrix.org)
- And then there's [#whatwg on Libera](ircs://irc.libera.chat/whatwg). It's bridged on Matrix at [#whatwg:libera.chat](https://matrix.to/#/#whatwg:libera.chat), but as of yesterday the bridge was still buggy.

If we decide we do want to keep the Libera channel, it would be interesting to [plumb it into the Matrix room](https://matrix-org.github.io/matrix-appservice-irc/latest/irc_operators.html#portals-and-plumbed).

**Comment by foolip** [comment]
Things just got bizarre:
https://twitter.com/tabatkins/status/1403490330985439235

I was also banned. I had a hard time caring about the original issue, feeling like maybe it was just maintainers failing to get along and being childish, like some ffmpeg fork drama.

But this makes me quite unimpressed, and my personal preference would be that we save any logs we can find, archive those on whatwg.org, and then stop linking to the Freenode channel.

**Comment by annevk** [comment]
I created https://github.com/whatwg/whatwg.org/pull/369 for the latter. Per https://freenode.logbot.info/_shutdown we should be able to get our logs for IRC at the end of this month and host them somewhere.

**Comment by globau** [comment]
> Per https://freenode.logbot.info/_shutdown we should be able to get our logs for IRC at the end of this month and host them somewhere.

For _reasons_ I've chosen to shutdown logbot early.  You can grab your #whatwg logs from https://freenode.logbot.info/static/exports/freenode/whatwg.gz (32.7 MB).  This URL is likely to change once I find a better home for the logs.

**Comment by MayeulC** [comment]
@globau wouldn't a torrent and/or IPFS make for a perfect, long-term home for these logs? Since they aren't going to change anytime soon, a content-addressed system looks perfect to me. That way, even if you shut down the web mirror after a while, interested parties might still seed it.

This is also something the Internet Archive might be interested in.

Thanks for your logging service ;)

**Comment by devkadirselcuk** [comment]
I think it looks very nice, clean and polite, and of course the evaluation
of the slzs is very important, thanks :-)

15 Haz 2021 Sal 17:24 tarihinde Mayeul Cantan ***@***.***>
şunu yazdı:

**Comment by annevk** [comment]
@bakkot perhaps we could convert the IRC logs to something Matrix-compatible and have your log host those as well? The source files do not seem to be too different.

**Comment by bakkot** [comment]
Sure, that should be pretty straightforward. I can't promise I'll get to it right away though.

**Comment by bakkot** [comment]
Alright, done: https://matrixlogs.bakkot.com/irc-whatwg/2006-12-01#L13. My log host is not as full-featured as traditional IRC log hosts like logbot, mind - in particular there's no search, since it's hosted on github pages and therefore strictly static content.

Let me know if there's other historical web platform channels you think it would make sense to add. I [automated it](https://github.com/bakkot/matrix-archive-bot/blob/0cf9aef05839ad1d27b44d5c7de1a6ca79860371/irc-logs-to-fake-json.js) (obviously), so it's straightforward to add new channels.

**Comment by annevk** [comment]
Thanks @bakkot, that looks great!
